Assets/LeapMotion/Scripts/HandRepresentation.cs

```diff
@@ -1,42 +1,80 @@
-﻿using UnityEngine;
-using System;
-using System.Collections;
-using System.Collections.Generic;
-
-using Leap;
+﻿using System.Collections.Generic;
 
 namespace Leap.Unity {
-  public abstract class HandRepresentation
-  {
+  /**
+   * HandRepresentation is a container class that facillitates the IHandModel lifecycle
+   * @param parent The HandPool which creates HandRepresentations
+   * @param handModel the IHandModel to be paired with Leap Hand data.
+   * @param hand The Leap Hand data to paired with an IHandModel
+   */
+  public class HandRepresentation {
+    HandPool parent;
     public int HandID { get; private set; }
     public int LastUpdatedTime { get; set; }
     public bool IsMarked { get; set; }
-    public Chirality RepChirality { get; protected set;}
-    public ModelType RepType { get; protected set;}
+    public Chirality RepChirality { get; protected set; }
+    public ModelType RepType { get; protected set; }
     public Hand MostRecentHand { get; protected set; }
+    public Hand PostProcessHand { get; set; }
+    public HandPool.ModelGroup Group { get; set; }
+    public List<IHandModel> handModels;
 
-    public HandRepresentation(int handID, Hand hand, Chirality chirality, ModelType modelType) {
-      HandID = handID;
+    public HandRepresentation(HandPool parent, Hand hand, Chirality repChirality, ModelType repType) {
+      this.parent = parent;
+      HandID = hand.Id;
+      this.RepChirality = repChirality;
+      this.RepType = repType;
       this.MostRecentHand = hand;
-      this.RepChirality = chirality;
-      this.RepType = modelType;
+      this.PostProcessHand = new Hand();
+    }
 
+    /** To be called if the HandRepresentation no longer has a Leap Hand. */
+    public void Finish() {
+      if (handModels != null) {
+        for (int i = 0; i < handModels.Count; i++) {
+          handModels[i].FinishHand();
+          parent.ReturnToPool(handModels[i]);
+          handModels[i] = null;
+        }
+      }
+      parent.RemoveHandRepresentation(this);
     }
 
-    /**
-    * Notifies the representation that a hand information update is available
-    * @param hand The current Leap.Hand.
-    * @param modelType Filters for a type of hand model, for example, physics or graphics hands.
-    */
-    public virtual void UpdateRepresentation(Hand hand) {
-      MostRecentHand = hand;
+    public void AddModel(IHandModel model) {
+      if (handModels == null) {
+        handModels = new List<IHandModel>();
+      }
+      handModels.Add(model);
+      if (model.GetLeapHand() == null) {
+        model.SetLeapHand(MostRecentHand);
+        model.InitHand();
+        model.BeginHand();
+        model.UpdateHand();
+      }
+      else {
```

Comment:
This else clause breaks our super strict uber-ultra magical coding conventions, it should be on the same line as the closing bracket as the starting if statement.